Yatima/FromLean.lean

```diff
@@ -159,50 +169,56 @@ mutual
     | none   => throw s!"'{nam}' not found in '{lvls}'"
   | .mvar .. => throw "Unfilled level metavariable"
 
+mutual
+
   partial def toYatimaRecursorRule
     (ctorCid : ConstCid) (rules : Lean.RecursorRule) :
-      EnvM RecursorRule := do
+      ToYatimaM RecursorRule := do
     let rhs ← toYatimaExpr [] rules.rhs
     let rhsCid ← exprToCid rhs
     addToEnv $ .expr_cache rhsCid rhs
     return ⟨ctorCid, rules.nfields, rhsCid⟩
 
   partial def toYatimaExpr (levelParams : List Lean.Name) :
-      Lean.Expr → EnvM Expr
+      Lean.Expr → ToYatimaM Expr
     | .bvar idx _ => return .var "" idx
     | .sort lvl _ => do
       let univ ← toYatimaUniv levelParams lvl
       let univCid ← univToCid univ
       addToEnv $ .univ_cache univCid univ
       return .sort univCid
-    | .const nam lvls _ => do
-      match (← read).find?' nam with
-      | some leanConst =>
-        let const ← toYatimaConst leanConst
-        let constId ← constToCid const
-        addToEnv $ .const_cache constId const
-        let univs ← lvls.mapM $ toYatimaUniv levelParams
-        let univsCids ← univs.mapM univToCid
-        (univsCids.zip univs).forM fun (univCid, univ) =>
-          addToEnv $ .univ_cache univCid univ
-        return .const nam constId univsCids
-      | none => throw s!"Unknown constant '{nam}'"
+    | .const nam lvls _ => do match (← get).constSubst with
+      | some varName => return .var varName (← get).nBinders
```

Comment:
Shouldn't we reset `constSubst` back to `None` once we're done with it? Either here or below